common/crypto_utils/avalon_crypto_utils/signature.py

```diff
@@ -269,21 +270,17 @@ def generate_client_signature(
             input_json_params, session_key,
             session_iv, worker.encryption_key, data_key, data_iv)
 
-        if input_json_params["requesterNonce"] and \
-                is_valid_hex_str(input_json_params["requesterNonce"]):
-            nonce = crypto.string_to_byte_array(
-                input_json_params["requesterNonce"])
-        else:
+        if "requesterNonce" in input_json_params and \
+                len(input_json_params["requesterNonce"]) == 0:
             # [NO_OF_BYTES] 16 BYTES for nonce.
             # This is the recommendation by NIST to
             # avoid collisions by the "Birthday Paradox".
-            nonce = crypto.random_bit_string(NO_OF_BYTES)
+            input_json_params["requesterNonce"] = secrets.token_hex(16)
 
-        request_nonce_hash = crypto.compute_message_hash(nonce)
-        nonce_hash = (crypto.byte_array_to_base64(request_nonce_hash)).encode(
-            'UTF-8')
         hash_string_1 = self.__calculate_hash_on_concatenated_string(
-            input_json_params, nonce_hash)
+            input_json_params, input_json_params["requesterNonce"].encode(
+                'UTF-8'
+            ))
```

Comment:
Shouldn't we use hash of nonce ? Trusted compute spec v1.1 mentions "Generate a random number and store its hash in requesterNonce" in section 6.6.1 Request Hash Calculation and Encryption

Reply:
requesterNonce is a random string generated by the participant. At line 278, creating random hex string.

Reply:
@manojgop Earlier steps are not correct. we need to send the random string in requesterNonce and hash will calculated on the string and used in request hash calculation.

Reply:
Need to modify the function `__calculate_hash_on_concatenated_string()` . Update the function param from nonce_hash to nonce. Also the function `__calculate_hash_on_concatenated_string()`  should be modified to calculate nonce hash from nonce as per spec 6.6.1. This function takes nonce (instead of noce_hash) as argument now. Calculate the nonce_hash within this function and concatenate it for final hash value calculation.

- Generate a random number and store its hash in requesterNonce.

- Calculate a hash value of the string concatenating the following values: requesterNonce, workOrderId, workerId, workloadId, and requesterId.

Reply:
Those steps are taken in out side of __calculate_hash_on_concatenated_string(), in generate_client_signature().

Reply:
requesterNonce is a random string generated by the participant. It is used to calculate a hash of this work order request. generate_client_signature() is shouldn't override that, if requesterNonce is passed in the request use that otherwise generate random string and use it in generate_client_signature(), it is taken care in that function.

Reply:
@manojgop I have review comments. Please review.